drivers/pse/si3474/si3474.c

```diff
@@ -0,0 +1,383 @@
+		res = i2c_burst_read(cfg->i2c.bus, dev_address, PORT3_VOLTAGE_LSB_REG, &buff[0],
+				     VOLTAGE_SIZE_BUFFER);
+		if (res != 0) {
+			LOG_WRN("Si3474 reading voltage on channel %d failed!", channel);
+		} else {
+			*val = ((uint16_t)buff[1] << 8);
+			*val |= (uint16_t)buff[0];
+		}
+		break;
+
+	case CHANNEL_3:
+		res = i2c_burst_read(cfg->i2c.bus, dev_address, PORT4_VOLTAGE_LSB_REG, &buff[0],
+				     VOLTAGE_SIZE_BUFFER);
+		if (res != 0) {
+			LOG_WRN("Si3474 reading voltage on channel %d failed!", channel);
+		} else {
+			*val = ((uint16_t)buff[1] << 8);
+			*val |= (uint16_t)buff[0];
+		}
+		break;
+	}
+	return res;
+}
+
+int si3474_get_main_voltage(const struct device *dev, uint16_t *val)
+{
+	const struct si3474_config *cfg = dev->config;
+	uint16_t dev_address = cfg->i2c.addr;
+	uint8_t buff[VOLTAGE_SIZE_BUFFER];
+	int res = 0;
+
+	res = i2c_burst_read(cfg->i2c.bus, dev_address, VPWR_LSB_REG, &buff[0],
+			     VOLTAGE_SIZE_BUFFER);
+	if (res != 0) {
+		LOG_WRN("Si3474 reading main voltage failed!");
+	} else {
+		*val = ((uint16_t)buff[1] << 8);
+		*val |= (uint16_t)buff[0];
+	}
+	return res;
+}
+
+static int si3474_get_temperature(const struct device *dev, uint8_t *buff)
+{
+	const struct si3474_config *cfg = dev->config;
+	uint16_t dev_address = cfg->i2c.addr;
+	int res = 0;
+
+	res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, TEMPERATURE_REG, buff);
+	if (res != 0) {
+		LOG_WRN("Si3474 reading temperature failed!");
+		return res;
+	}
+	return res;
+}
+
+static int si3474_switch_on_channel(const struct device *dev, uint8_t channel)
+{
+	int res;
+	const struct si3474_config *cfg = dev->config;
+	int16_t dev_address = cfg->i2c.addr;
+
+	res = i2c_reg_write_byte(cfg->i2c.bus, dev_address, PB_POWER_ENABLE_REG, (1 << channel));
+	if (res != 0) {
+		LOG_WRN("Si3474 switching channel failed");
+		return res;
+	}
+
+	return res;
+}
+
+static int si3474_switch_off_channel(const struct device *dev, uint8_t channel)
+{
+	int res;
+	const struct si3474_config *cfg = dev->config;
+	int16_t dev_address = cfg->i2c.addr;
+
+	res = i2c_reg_write_byte(cfg->i2c.bus, dev_address, PB_POWER_ENABLE_REG,
+				 (1 << (channel + 4)));
+	if (res != 0) {
+		LOG_WRN("Si3474 switching channel failed");
+		return res;
+	}
+
+	return res;
+}
+
+static int si3474_init(const struct device *dev)
+{
+	const struct si3474_config *cfg = dev->config;
+	int res;
+#if CONFIG_SI3474_TRIGGER
+	res = si3474_init_interrupt(dev);
+	if (res != 0) {
+		LOG_ERR("Configuring Si3474 interrupt callback failed");
+		return -ENODEV;
+	}
+#endif
+	if (device_is_ready(cfg->i2c.bus) == false) {
+		LOG_ERR("I2C bus is not ready");
+		return -ENODEV;
+	}
+	uint32_t i2c_cfg = I2C_SPEED_SET(I2C_SPEED_FAST) | I2C_MODE_CONTROLLER;
+	res = i2c_configure(cfg->i2c.bus, i2c_cfg);
+	if (res != 0) {
+		LOG_ERR("Configuring Si3474 speed failed");
+		return -ENODEV;
+	}
+
+	res = si3474_init_ports(dev);
+	if (res != 0) {
+		LOG_ERR("Configuring Si3474 interrupt and oss pins failed");
```

Comment:
I dont know what oss is 😅 If it is an abbreviation maybe it should be capital letters.
Edit: Ah, over supply signal. Maybe it's ok since it's in the binding

Reply:
I will correct description to Over Supply Signal ;)